### DISCUSSION TITLE
Is there a good way to compose translations?

### DISCUSSION THREAD

**Discussion by talyh** [vinissimus/next-translate#873] [Q&A] [answered]
I'm looking to understand whether there's a way to compose translations.

For instance, say we have this in `common.json`:
```json
{
  "cats": "cats"
}
```
then we want to have this in a `specific-page.json`:
```json
{
   "blurb": "We love cats"
}
```

**Is there a good way to leverage that `cats` keyword in the larger blurb?**

### Addendum

I think we can change `"blurb"` so it's ` "blurb": "We love {{cats}}"`, and then pass it in as 
```javascript
t("specific-page:blurb", {cats: t("common:cat")}
```
, but feels very verbose especially when we apply to more real life keys (not to mention the issue with pluralization, as ideally the `cats` key would be singular and we'd pluralize in context, but leaving that aside for simplicity here).
So looking for.a better way if there's a pattern of composition we could/should be using

**Comment by aralroca** [comment, marked as answer]
For now there is no better way implemented

**Reply by talyh** [reply]
Thanks for confirming @aralroca!
Are you aware of an issue where this is already tracked? Or, is it worth opening a new issue for it?

**Reply by aralroca** [reply]
I think is similar to this issue: https://github.com/vinissimus/next-translate/issues/872

**Reply by talyh** [reply]
Thank you, that sounds exactly right!  🙏🏻 
I couldn't find that issue when searching before.
